### ISSUE TITLE
Feature Request  - Internet Radio support

### ISSUE THREAD

**Issue by ripdog** [ludouzi/fooyin#84]
Internet radio shouldn't be super hard to implement, AFAIK FFMPEG handles it natively.

If I could make one request, please handle network errors as playback-ended events, so don't force playback stop when they happen. I have a favorite internet radio which seems to cause periodic errors in whatever player I use, so it would be great if I could simply use playlist-repeat to force fooyin to keep trying to play the stream.

**Comment by ludouzi** [comment]
Yeah, I've just had a quick look into it and it should be fairly trivial to implement by passing the stream url to FFmpeg. I'll add it to the roadmap.

I can add a setting to optionally skip to the next stream in case of errors. I'll probably add that in a bit as a dialog is currently shown when a local track can't be found, pausing playback temporarily.

**Comment by LukeZBaker** [comment]
Many audio players make the mistake of making it non-trivial to add stations. 

When implementing internet radio, it really would help using a community database for stations: radio-browser.info is probably the best.

**Comment by giantplaceholder** [comment]
> Yeah, I've just had a quick look into it and it should be fairly trivial to implement by passing the stream url to FFmpeg. I'll add it to the roadmap.
> 
> I can add a setting to optionally skip to the next stream in case of errors. I'll probably add that in a bit as a dialog is currently shown when a local track can't be found, pausing playback temporarily.

Any chance you can take some time and implement this? Lack of streaming support is basically the only reason I keep launching deadbeef any time I'd like to listen to some Nectarine.FM

**Comment by ludouzi** [comment]
> Any chance you can take some time and implement this? 

There is a good chance I will implement this for 0.11.0. The only planned feature for that release is the spectrum, which is already half done.

**Comment by bassstorm** [comment]
@ludouzi would you consider making stream playback architecture flexible enough, so that a) user could tweak tags for streams (obviously values cannot be attached to stream directly, but saved as part of playlist -- or database -- instead), and b) a plugin could manage playlist with stream entries?

With regards to the latter, my long standing dream is to have fully functional Bandcamp plugin that able to parse Bandcamp JSON model so that each streaming track is tagged properly for scrobbling, but also it could refresh URLs in advance or by user request. I mean, such a plugin as a task in itself, but plugin API should support such integration endpoints.
I myself have implemented a half-cooked specialized player for Bandcamp flow :-), but seeing something like that part of Fooyin would resolve many other basics which I'd probably never could address.
